javascript/javascript3/week1/homework/weather-app/weather-app.js

```diff
@@ -0,0 +1,93 @@
+const cityInput = document.getElementById("cityInput");
+const todayDate = document.getElementById("todayDate");
+const cityName = document.getElementById("cityName");
+const temp = document.getElementById("temp");
+const tempFeelsLike = document.getElementById("tempFeelsLike");
+const icon = document.getElementById("icon");
+const windSpeed = document.getElementById("windSpeed");
+const cloudy = document.getElementById("cloudy");
+const sunrise = document.getElementById("sunrise");
+const sunset = document.getElementById("sunset");
+
+function showWeather(content) {
+  const today = new Date();
+  todayDate.innerHTML = today.toLocaleDateString();
+  cityName.innerHTML = content.name;
+  icon.src = `http://openweathermap.org/img/wn/${content.weather[0].icon}@2x.png`;
+  temp.innerHTML = `${Math.floor(content.main.temp)}&deg;C`;
+  tempFeelsLike.innerHTML = `Feels like ${Math.floor(
+    content.main.feels_like
+  )}&deg;C, ${content.weather[0].description}`;
+  windSpeed.innerHTML = `Wind: ${Math.floor(content.wind.speed)}m/s`;
+  cloudy.innerHTML = `Humidity: ${content.clouds.all}%`;
+  let sunriseFormatted = new Date(content.sys.sunrise * 1000);
+  sunrise.innerHTML = `Sunrise: ${sunriseFormatted.toLocaleTimeString()}`;
+  let sunsetFormatted = new Date(content.sys.sunset * 1000);
+  sunset.innerHTML = `Sunset: ${sunsetFormatted.toLocaleTimeString()}`;
+  function initMap() {  //doesn't work :(
```

Comment:
It's better to write this function outside of the "showWeather" function and through this function just call it like this:
" initMap(content.coord.lat,content.coord.lon); "

The code of "initMap" should modify like this:
function initMap(latitude, longitude) {
    // The location of city
    const city = { lat: latitude, lng: longitude };
    // The map, centered at current location
    const map = new google.maps.Map(document.getElementById("map"), {
      zoom: 6,
      center: city,
    });
    // The marker, positioned at current city
    const marker = new google.maps.Marker({
      position: city,
      map: map,
    });
  }